client/src/components/dashboard/Profile/Preferences/common/RepoList.js

```diff
@@ -171,20 +173,12 @@ class RepoList extends React.Component {
 
     // GITLAB VALIDATIONS:
     if (label === 'https://gitlab.com/') {
-      // CHALLENGE!!! anyone up for the challenge of writing a regex that covers all of this?
-      // I was unable to, would def clean this up. -Pete
 
       if (
-      // GitLab naming conventions:
-      repo.slice(-4) === '.git' ||
-      repo.slice(-5) === '.atom' ||
-      repo.slice(-1) === '.' ||
-      repo.slice(0, 1) === '-' ||
-      namespace.slice(-4) === '.git' ||
-      namespace.slice(-5) === '.atom' ||
-      namespace.slice(0, 1) === '-' ||
-      namespace.slice(-1) === '.' ||
-      !Validator.matches(item, /[\d\w-.]+\/[\d\w-.]+\/?/)
+        // GitLab naming conventions:
+        /(^-)|(\.((git)|(atom))?$)/.test(repo) ||
+        /(^-)|(\.((git)|(atom))?$)/.test(namespace) ||
```

Comment:
@gwenf this is infinitely better, thank you! I had previously been using `startsWith` and `endsWith` instead of the series of `slice`s, but since those are ES6 methods they wouldn't run in all browsers and I was having trouble polyfilling it. This is better than either of those ways though.